src/reporter/coverage-report.xsl

```diff
@@ -252,19 +220,47 @@
       </xsl:non-matching-substring>
     </xsl:analyze-string>
   </xsl:variable>
-  <xsl:sequence select="$analyzed/node()[not(self::test:residue)]"/>
-  <xsl:variable name="residue" as="element(test:residue)?" select="$analyzed/test:residue"/>
-  <xsl:if test="$residue/@rest ne ''">
-    <!-- The last thing this template does is call itself.
-         Tail recursion prevents stack overflow. -->
-    <xsl:call-template name="test:output-lines">
-      <xsl:with-param name="line-number" select="$line-number + xs:integer($residue/@count) - 1" />
-      <xsl:with-param name="stylesheet-string" select="string($residue/@rest)" />
+
+  <xsl:iterate select="$regex-groups">

```

Comment:
How about a comment like "Iterate over the matching substrings, while keeping track of the corresponding line number and node."?